### ISSUE TITLE
UI Bug #2

### ISSUE THREAD

**Issue by JackOrez** [helgoboss/helgobox#1213]
Bug #2:
Could possibly be related to this?
- ReaLearn: #1157 Improved control via keyboard by not processing keyboard input when user is entering text (needs REAPER 7.23+)

In versions prior to 2.16.4, I could enter text into any text input field in Helgobox and then just click any of it's title bar's to release focus,
however now I have to click 'OK' to exit the 'Mapping' page or click completly outside of Helgobox to release the focus.

This is also problematic on the main page, for example if you click the VST preset dropdown and then try to release focus by clicking
the title bar, the dropdown will close but it still has focus.[Not good if you have mappings that use Computer Keyboard as input and presets names
that start with that key.] By the way, is there some technical reason why the user can't just click an empty spot within the UI in order to
release focus from controls?

**Comment by helgoboss** [comment]
I just tried this on macOS. The focus behavior didn't change there:

- Clicking the title bar doesn't make the text field lose focus
- Clicking on some other UI elements (but not all) makes the text field lose focus
- Pressing TAB or ENTER makes the text field lose focus. In a mapping panel, ENTER also closes the mapping panel.

When using keyboard sources with REAPER 7.23+, not making the text field lose focus will prevent the source from learning/firing, so there's that. But I think you didn't run into that yet.

But I know you use Windows. Trying that next.

**Comment by helgoboss** [comment]
I also tried on Windows. The focus behavior didn't change between 2.16.3 and 2.16.4 there either. Result:

- Clicking the title bar doesn't make the text field lose focus
- Clicking on some other UI elements (but only active ones, it seems) makes the text field lose focus
- Pressing TAB or ENTER doesn't have any effect

Okay, that is bad. But it didn't seem to change within ReaLearn. I also tried with an older REAPER version (7.17) and it still has this bad behavior. Maybe a Windows update?

I'll try to make it work at least with TAB or ENTER. This is what I always used.
